src/com/mambu/apisdk/services/OwnedEntityService.java

```diff
@@ -0,0 +1,189 @@
+ * deleting owned entities). Extending class must specify the type of entity they manage (their owned entity, example
+ * Document) the supported API methods and the set of parent entities that are supported for this owned entity.
+ * 
+ * For example, CommentsService can extend this class and would specify MambuEntity.COMMENT as the owned entity,
+ * supported API methods as GET and POST, and also a set of MambuEntities supported by the Comments API. The
+ * OwnedEntityService provides the implementation for the actual API calls
+ * 
+ * @author mdanilkis
+ * 
+ */
+
+public abstract class OwnedEntityService {
+
+	// Extended classes must specify the type of owned entity. Example, Document, Comment
+	abstract protected MambuEntity getOwnedEntity();
+
+	// Extended classes must specify supported parent classes
+	abstract protected Set<MambuEntity> getSupportedEntities();
+
+	// Extended classes must specify API methods supported by it
+	abstract protected Set<RequestExecutor.Method> getSupporteMethods();
+
+	// Service helper
+	protected ServiceExecutor serviceExecutor;
+
+	/***
+	 * Create a new service
+	 * 
+	 * @param mambuAPIService
+	 *            the service responsible with the connection to the server
+	 */
+	@Inject
+	public OwnedEntityService(MambuAPIService mambuAPIService) {
+		this.serviceExecutor = new ServiceExecutor(mambuAPIService);
+	}
+
+	/***
+	 * Get all owned entities for a given parent entity
+	 * 
+	 * @param parentEntity
+	 *            Mambu entity for which owned entities are retrieved. Example, Client, Branch
+	 * 
+	 * @param entityId
+	 *            entity id or encoded key. Example, client id for a Client.class
+	 * 
+	 * @return a list of owned entities for parentEntity
+	 * 
+	 * @throws MambuApiException
+	 */
+	public <T> List<T> getOwnedEntities(MambuEntity parentEntity, String entityId, Integer offset, Integer limit)
+			throws MambuApiException {
+		// Example: GET /api/clients/ABC123/comments ; GET /api/savings/ABC123/comments
+		// See MBU-8608 for more details
+
+		// Validate GET is supported
+		validateAPiIsSupported(Method.GET);
+
+		// Validate the API is used for the Supported entity
+		validateParentEntity(parentEntity);
+
+		// Get a list of owned entities for a parent entity
+		// Derived class provides the owned entity managed by it. Example, Comment, CustomFieldValue
+		MambuEntity ownedEntity = getOwnedEntity();
+		return serviceExecutor.getOwnedEntities(parentEntity, entityId, ownedEntity, offset, limit);
+
+	}
+
+	/**
+	 * Post new owned entity
+	 * 
+	 * @param parentEntity
+	 *            class for the entity for which comments are retrieved. Example, Client.class, Branch.class
+	 * @param parentEntityId
+	 *            entity id or encoded key for the parent
+	 * @param ownedEntity
+	 *            owned entity to post
+	 * @throws MambuApiException
+	 */
+	public <R, T> R postOwnedEntity(MambuEntity parentEntity, String parentEntityId, T ownedEntity)
```

Comment:
Can we rename this to "createOwnedEntity"?
Or is this also used for overriding owned entities? I suppose not, because then we'd need to have the owned entity ID in the URL.